avm/res/resources/deployment-script/tests/e2e/private-network/main.test.bicep

```diff
@@ -0,0 +1,96 @@
+targetScope = 'subscription'
+
+metadata name = 'Using Private Endpoints'
+metadata description = 'This instance deploys the module with access to a private network.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'avm-${namePrefix}-resources.deploymentscripts-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'rdsnet'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    storageAccountName: 'dep${namePrefix}st${serviceShort}'
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: location
+    azCliVersion: '2.9.1'
+    kind: 'AzureCLI'
+    retentionInterval: 'P1D'
+    cleanupPreference: 'Always'
+    lock: {
+      kind: 'None'
+    }
+    subnetResourceIds: [
+      nestedDependencies.outputs.subnetResourceId
+    ]
+    containerGroupName: 'dep-${namePrefix}-cg-${serviceShort}'
+    arguments: '-argument1 \\"test\\"'
+    environmentVariables: {
+      secureList: [
+        {
+          name: 'var1'
+          value: 'test'
+        }
+        {
+          name: 'var2'
+          secureValue: guid(deployment().name)
+        }
+      ]
+    }
+    managedIdentities: {
+      userAssignedResourcesIds: [
+        nestedDependencies.outputs.managedIdentityResourceId
+      ]
+    }
+    roleAssignments: [
+      {
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+        roleDefinitionIdOrName: 'Reader'
+      }
+    ]
```

Comment:
```suggestion
```

I'd opt to remvoe this as it's not technically required for private networking. The same for other parameter like the `environmentVariables` etc. if you agree.
The reasoning why I'm suggesting this is that a 'sanitized' example would show a user exactly what parameter they'd need to get this to work. If we also add other parameters a user could get the impression that they'd need to deploy e.g. this role assignment to get the example to work. I hope that makes sense.

Note, this updated requires a regeneration of the readme